### ISSUE TITLE
Is there way to compare R and Python implementations?

### ISSUE THREAD

**Issue by alexander-rakhlin** [dmlc/xgboost#604]
Hi all,

It all started as I ported a model from R to Python and noticed performance worsened. I tried then to compare a simplistic model on a toy data to exclude all sources of incomparability. Without avail: Python and R versions of Xgboost produce slightly different models, it is noticeable at the training time. What is more strange, if seed is not set, R implementation trains model differently every run. Python implementation does not depend on seed.

Both are current releases installed via CRAN and pip. This behavior remains the same on Ubuntu and Windows. Can I still run R and Python Xgboost in a comparable way?

**Comment by tqchen** [comment]
this is due to random seed. Normally you should not relies on random seed to get good perfmance, since this can mean overfitting. In other sense, all bindings are fully compatible
